return (function(...)
	local k = {
		"\048\109\114\070\048\084\067\116\065\068\086\075\089\068\088\101\048\084\119\114\121\072\079\061",
		"\090\108\108\086\079\118\090\106\052\118\076\116",
		"\119\075\115\090\043\100\056\043\071\085\072\082\100\070\076\061",
		"\084\111\100\054\048\068\097\061",
		"\065\105\107\106\079\072\104\111\104\068\112\072\082\106\051\043\048\053\061\061",
		"\076\078\075\090\069\070\080\061",
		"\072\086\048\111\070\084\089\055\069\111\117\090\106\111\108\053\116\054\053\108\101\078\061\061",
		"\117\085\103\074",
		"\069\076\087\086\075\108\054\080\073\087\107\105\056\055\065\118\068\055\122\073\076\053\061\061",
		"\075\103\084\057\097\079\061\061",
		"\121\072\067\070\089\068\088\057",
		"\114\084\069\088\110\078\080\061",
		"\084\111\100\057\120\053\061\061",
		"\065\068\088\053\120\068\086\083",
		"\075\114\109\106\050\085\054\061",
		"\073\077\109\107\055\049\101\061",
		"\099\105\065\120\104\052\061\061",
		"\048\109\073\116\117\072\076\061",
		"\071\070\104\104\078\071\066\102\053\043\083\068\113\089\099\090\115\052\061\061",
		"\085\100\116\114\098\118\115\043",
		"",
		"\121\109\103\051\117\072\048\114",
		"\055\071\085\118\116\087\070\052\086\078\061\061",
		"\101\100\048\114\089\109\119\048\100\050\072\088",
		"\056\102\071\076\072\107\067\101\080\122\087\119\114\120\083\111\097\101\055\087\113\053\061\061",
		"\056\115\116\052\100\052\061\061",
		"\108\097\110\114\110\078\100\053\105\053\061\061",
		"\103\055\107\050\065\106\048\116\121\080\086\075\089\068\073\080",
		"\120\055\118\118\121\052\061\061",
		"\118\051\100\074\117\116\056\089\085\070\067\105",
		"\114\052\069\108\111\079\065\119\106\078\061\061",
		"\082\105\084\100\119\105\056\043\117\053\061\061",
		"\109\079\057\069\078\098\088\101\111\051\068\076",
		"\067\055\103\101\079\055\118\050\117\085\067\070\048\068\097\061",
		"\048\101\112\069\048\087\100\087\048\122\086\110\087\109\086\081\086\068\080\061",
		"\084\101\121\061",
		"\120\057\114\101\048\079\061\061",
		"\065\085\107\102\117\085\087\061",
		"\067\067\086\110\116\112\067\078\087\072\118\048\081\105\084\102\056\116\113\078\107\052\061\061",
		"\122\117\072\082\071\110\112\090\115\116\050\053\080\081\099\113\073\112\097\071\107\078\061\061",
		"\065\085\107\069\089\053\061\061",
		"\065\055\100\070\089\072\086\053\120\068\086\114",
		"\118\070\049\113\078\048\074\077",
		"\048\055\107\051\048\079\061\061",
		"\121\085\107\050\121\057\104\061",
		"\121\055\103\101\117\068\103\101\120\084\067\118\120\109\073\114",
		"\087\068\114\110\076\049\102\115\080\101\075\074\114\088\052\061",
		"\066\049\116\101\072\098\051\103",
		"\056\109\065\054\089\111\103\122\077\085\111\054\077\087\114\085",
		"\080\108\105\101\111\111\090\100\053\102\099\101",
		"\121\109\107\074\048\085\100\051",
		"\117\068\107\101\089\078\061\061",
		"\088\071\107\070\053\098\113\108\118\053\061\061",
		"\084\111\100\051\048\084\067\118\065\085\107\102\117\085\087\061",
		"\084\111\100\050\117\109\067\114\077\078\061\061",
	}
	for C, y in ipairs({ { 1, 55 }, { 1, 4 }, { 5, 55 } }) do
		while y[1] < y[2] do
			k[y[1]], k[y[2]], y[1], y[2] = k[y[2]], k[y[1]], y[1] + 1, y[2] - 1
		end
	end
	local function C(C)
		return k[C + 59123]
	end
	do
		local C = string.sub
		local y = table.insert
		local M = math.floor
		local N = string.char
		local s = type
		local e = k
		local w = {
			H = 55,
			J = 46,
			["\047"] = 2,
			g = 21,
			j = 4,
			["\055"] = 54,
			i = 3,
			["\057"] = 39,
			y = 28,
			V = 13,
			L = 8,
			X = 57,
			G = 7,
			P = 36,
			["\050"] = 41,
			p = 60,
			C = 17,
			I = 49,
			N = 0,
			W = 20,
			n = 10,
			["\049"] = 31,
			l = 59,
			a = 56,
			O = 16,
			t = 47,
			A = 29,
			k = 5,
			U = 6,
			q = 63,
			["\048"] = 25,
			["\052"] = 32,
			v = 33,
			["\051"] = 45,
			K = 40,
			M = 30,
			T = 23,
			D = 22,
			z = 35,
			r = 37,
			x = 24,
			F = 50,
			f = 34,
			Z = 1,
			B = 11,
			o = 53,
			S = 43,
			h = 12,
			["\053"] = 48,
			E = 51,
			d = 61,
			b = 58,
			Q = 19,
			e = 52,
			R = 14,
			["\056"] = 18,
			s = 15,
			w = 9,
			["\043"] = 42,
			["\054"] = 44,
			c = 62,
			u = 27,
			m = 38,
			Y = 26,
		}
		local g = table.concat
		local G = string.len
		for k = 1, #e, 1 do
			local Y = e[k]
			if s(Y) == "\115\116\114\105\110\103" then
				local s = G(Y)
				local i = {}
				local v = 1
				local p = 0
				local z = 0
				while v <= s do
					local k = C(Y, v, v)
					local e = w[k]
					if e then
						p = p + e * 64 ^ (3 - z)
						z = z + 1
						if z == 4 then
							z = 0
							local k = M(p / 65536)
							local C = M((p % 65536) / 256)
							local s = p % 256
							y(i, N(k, C, s))
							p = 0
						end
					elseif k == "\061" then
						y(i, N(M(p / 65536)))
						if v >= s or C(Y, v + 1, v + 1) ~= "\061" then
							y(i, N(M((p % 65536) / 256)))
						end
						break
					end
					v = v + 1
				end
				e[k] = g(i)
			end
		end
	end
	return (function(k, M, N, s, e, w, g, o, v, b, z, y, G, S, i, Y, p)
		z, G, S, b, v, p, y, o, Y, i =
			function(k)
				local C, y = 1, k[1]
				while y do
					Y[y], C = Y[y] - 1, 1 + C
					if Y[y] == 0 then
						Y[y], G[y] = nil, nil
					end
					y = k[C]
				end
			end, {}, function(k, C)
				local M = p(C)
				local N = function(...)
					return y(k, { ... }, C, M)
				end
				return N
			end, function(k)
				Y[k] = Y[k] - 1
				if Y[k] == 0 then
					Y[k], G[k] = nil, nil
				end
			end, 0, function(k)
				for C = 1, #k, 1 do
					Y[k[C]] = Y[k[C]] + 1
				end
				if N then
					local y = N(true)
					local M = e(y)
					M[C(-59119)], M[C(-59106)], M[C(-59115)] = k, z, function()
						return 3198946
					end
					return y
				else
					return s({}, {
						[C(-59106)] = z,
						[C(-59119)] = k,
						[C(-59115)] = function()
							return 3198946
						end,
					})
				end
			end, function(y, N, s, e)
				local K, p, V, S, Y, a, r, q, x, m, T, u, c, n, X, g, z, Q, D, j, f, h, v, A, E, Z, F
				while y do
					if y < 9024172 then
						if y < 5553324 then
							if y < 618899 then
								if y < 359729 then
									if y < 321040 then
										j, a = T(f, j)
										y = j and 5456038 or 8118520
									else
										Y = N[1]
										y = G[s[1]]
										p = y
										v = N[2]
										y = p[v]
										y = y and 16096236 or 559633
									end
								else
									if y < 469138 then
										Q = f
										y = Q
										T[Q] = y
										Q = nil
										y = 8609031
									else
										c = -198117
										q = 663413
										y = {}
										r = 751287
										G[s[2]] = y
										g = G[s[3]]
										X = 35184371337545
										S = r + X
										x = 663668
										z = g
										g = v % S
										G[s[4]] = g
										X = x - q
										q = 11744
										u = -198118
										x = 11746
										y = 12613505
										r = v % X
										X = x - q
										S = r + X
										m = -681588
										x = C(-59108)
										G[s[5]] = S
										X = k[x]
										q = 205734
										x = C(-59111)
										r = X[x]
										x = 205900
										X = r(Y)
										r = C(-59098)
										p[v] = r
										r = x - q
										q = 681589
										x = q + m
										q = X
										m = c - u
										c = m
										m = 0
										u = c < m
										m = x - c
									end
								end
							else
								if y < 4398780 then
									if y < 3859096 then
										y = 11503455
										X = nil
										r = nil
										z = nil
									else
										p = 615219
										Y = G[s[1]]
										v = 615219
										g = #Y
										Y = v - p
										y = g == Y
										y = y and 11536736 or 13479324
									end
								else
									if y < 5225836 then
										a = 425805
										S = 894428
										z = -894396
										v = G[s[3]]
										T = -132708
										p = z + S
										Y = v % p
										m = 891710
										z = G[s[4]]
										c = 891708
										X = G[s[2]]
										q = m - c
										u = -132695
										c = u - T
										f = G[s[3]]
										j = -425773
										T = f - Y
										f = j + a
										u = T / f
										m = c - u
										x = q ^ m
										r = X / x
										X = -589746
										S = z(r)
										r = 4294377550
										z = r - X
										p = S % z
										X = 92643
										r = 92645
										S = r - X
										z = S ^ Y
										u = 1038912
										m = -948797
										v = p / z
										z = G[s[4]]
										q = -948796
										x = q - m
										X = v % x
										m = 4294163527
										q = 803769
										x = q + m
										r = X * x
										S = z(r)
										z = G[s[4]]
										Y = nil
										q = 1046848
										r = z(v)
										X = 257190
										p = S + r
										r = -191654
										S = r + X
										T = 1038656
										z = p % S
										x = 1112384
										m = 671221
										c = 670965
										r = p - z
										X = x - q
										S = r / X
										x = -112035
										q = -112291
										X = x - q
										r = z % X
										x = z - r
										q = m - c
										c = -824250
										X = x / q
										m = 824506
										z = nil
										q = m + c
										x = S % q
										v = nil
										m = S - x
										c = u - T
										S = nil
										y = 13479324
										q = m / c
										p = nil
										m = { r, X, x, q }
										q = nil
										G[s[1]] = m
										x = nil
										r = nil
										X = nil
									else
										Z = 11941447545306
										n = 296553
										A = C(-59089)
										y = 290824
										u = j
										E = -1048192
										F = C(-59076)
										K = 21099669704258
										h = Z - n
										Q = p(F, h)
										V = v[Q]
										D = K - E
										n = p(A, D)
										Z = v[n]
										K = -441326
										h = r[Z]
										A = C(-59080)
										E = 28299329839841
										D = K + E
										u = nil
										n = p(A, D)
										Z = v[n]
										n = C(-59071)
										D = 24846070056571
										K = -250856
										F = h[Z]
										A = D - K
										Z = p(n, A)
										h = v[Z]
										Q = F[h]
										a[V] = Q
										a = nil
									end
								end
							end
						else
							if y < 7089721 then
								if y < 7008312 then
									if y < 5893896 then
										u = nil
										m = nil
										y = c
										c = nil
										S = y
										y = 13832102
									else
										z = 642036
										S = -641834
										v = G[s[3]]
										p = z + S
										Y = v * p
										y = 7015638
										z = -995774
										p = 996031
										v = p + z
										g = Y % v
										G[s[3]] = g
									end
								else
									if y < 7028586 then
										S = -424940
										z = 424941
										v = G[s[3]]
										p = z + S
										Y = v ~= p
										y = Y and 4729488 or 7007958
									else
										Q = -673077
										a = -339307
										V = 9582364807239
										f = C(-59103)
										m = q
										j = a + V
										y = C(-59091)
										F = 14501326551542
										T = p(f, j)
										u = v[T]
										a = C(-59104)
										y = c[y]
										y = y(c, u)
										V = Q + F
										j = p(a, V)
										f = v[j]
										u = y
										T = u[f]
										y = T == r
										y = y and 5557929 or 15546331
									end
								end
							else
								if y < 8360153 then
									if y < 8072240 then
										v = nil
										S = nil
										g = {}
										q = nil
										x = nil
										X = nil
										c = nil
										p = nil
										y = k[C(-59070)]
										z = nil
										r = nil
										m = nil
									else
										Q = C(-59069)
										h = 18891990726507
										T = C(-59118)
										u = k[T]
										Z = 537265
										F = h - Z
										V = p(Q, F)
										Z = 831546
										a = v[V]
										h = 24104092895210
										j = r[a]
										F = h - Z
										Q = C(-59112)
										V = p(Q, F)
										a = v[V]
										f = j[a]
										a = -671029
										V = 671029
										j = a + V
										T = u(f, c, j)
										a = C(-59102)
										f = C(-59078)
										F = 2490082687773
										Z = -766200
										Q = 593394
										h = 23534990763174
										T = k[f]
										V = Q + F
										j = p(a, V)
										f = v[j]
										u = T[f]
										Q = C(-59117)
										f = 0.1
										T = u(f)
										T = C(-59118)
										u = k[T]
										F = h - Z
										V = p(Q, F)
										a = v[V]
										Z = 13085017285117
										Q = C(-59094)
										j = r[a]
										h = -94819
										F = h + Z
										V = p(Q, F)
										Z = 34526148551619
										a = v[V]
										Q = C(-59086)
										f = j[a]
										h = -288297
										a = 781433
										V = 781432
										j = a - V
										T = u(f, c, j)
										T = C(-59118)
										u = k[T]
										F = h + Z
										V = p(Q, F)
										Q = C(-59079)
										h = 30499493737306
										a = v[V]
										Z = 197734
										j = r[a]
										F = h - Z
										V = p(Q, F)
										h = -669685
										a = v[V]
										f = j[a]
										Z = 33888326824765
										a = 696118
										Q = C(-59095)
										V = 696118
										j = a - V
										T = u(f, x, j)
										T = C(-59118)
										u = k[T]
										F = h + Z
										Z = 8199343060220
										V = p(Q, F)
										Q = C(-59110)
										a = v[V]
										h = -268072
										j = r[a]
										F = h + Z
										V = p(Q, F)
										a = v[V]
										f = j[a]
										V = -108963
										a = -108962
										Q = -424621
										j = a - V
										T = u(f, x, j)
										a = C(-59109)
										f = C(-59078)
										F = 27908475871284
										T = k[f]
										V = Q + F
										j = p(a, V)
										f = v[j]
										u = T[f]
										y = 16417782
										T = u()
									end
								else
									if y < 8508283 then
										j = -410220
										f = 410221
										g = f + j
										f = #T
										Q = -648948
										y = S(g, f)
										f = y
										y = r(T, f)
										F = 648949
										j = y
										y = G[c]
										V = Q + F
										f = nil
										a = j - V
										g = X(a)
										y[j] = g
										y = 16081899
										j = nil
									else
										Q = not V
										f = f + a
										g = f <= j
										g = Q and g
										Q = f >= j
										Q = V and Q
										g = Q or g
										Q = 406540
										y = g and Q
										g = 14524860
										y = y or g
									end
								end
							end
						end
					else
						if y < 13520505 then
							if y < 11918404 then
								if y < 11452018 then
									if y < 9122907 then
										T = C(-59074)
										u = k[T]
										V = C(-59085)
										V = X[V]
										a = { V(X) }
										V = { u(M(a)) }
										y = 290824
										T = V[1]
										j = V[3]
										f = V[2]
									else
										q, c = X(x, q)
										y = q and 7064792 or 13832102
									end
								else
									if y < 11527298 then
										g = { v }
										y = k[C(-59084)]
									else
										z = 453294
										S = -453253
										v = G[s[2]]
										p = z + S
										Y = v * p
										z = 28116788740169
										p = -1022464
										v = p + z
										p = -692917
										g = Y + v
										v = 35184371395915
										Y = v - p
										y = g % Y
										G[s[2]] = y
										p = 424941
										Y = G[s[3]]
										y = 7007958
										z = -424940
										v = p + z
										g = Y ~= v
									end
								end
							else
								if y < 12782374 then
									if y < 12359640 then
										Q = C(-59108)
										x = m
										V = k[Q]
										Q = C(-59082)
										a = V[Q]
										V = a(Y, x)
										a = G[s[6]]
										Q = a()
										x = nil
										j = V + Q
										h = 148645
										V = -204902
										a = -204646
										f = j + r
										F = 148646
										j = a - V
										T = f % j
										y = 12613505
										r = T
										j = p[v]
										Q = F - h
										V = r + Q
										a = z[V]
										f = j .. a
										p[v] = f
									else
										T = not u
										m = m + c
										x = m <= q
										x = T and x
										T = m >= q
										T = u and T
										x = T or x
										T = 12228981
										y = x and T
										x = 621074
										y = y or x
									end
								else
									if y < 13069693 then
										j = i()
										a = i()
										X = nil
										n = nil
										X = C(-59087)
										y = {}
										f = i()
										G[f] = y
										y = o(4039803, { f, x, q, z })
										z = b(z)
										T = nil
										u = nil
										S = nil
										G[j] = y
										F = C(-59119)
										V = {}
										r = nil
										y = {}
										g = C(-59073)
										G[a] = y
										Z = C(-59120)
										y = k[g]
										h = G[a]
										Q = { [F] = h, [Z] = n }
										z = C(-59077)
										g = y(V, Q)
										u = 5410750888869
										v = g
										m = nil
										y = o(327690, { a, f, c, x, q, j })
										q = b(q)
										a = b(a)
										f = b(f)
										q = 15568658728159
										x = b(x)
										p = y
										j = b(j)
										c = b(c)
										T = 714742
										m = 904252
										y = k[z]
										x = q - m
										m = C(-59121)
										r = p(X, x)
										x = C(-59075)
										S = v[r]
										z = C(-59091)
										z = y[z]
										z = z(y, S)
										y = nil
										S = y
										X = k[x]
										c = u - T
										q = p(m, c)
										x = v[q]
										r = X[x]
										q = C(-59072)
										u = 30994502972237
										c = 654708
										m = c + u
										x = p(q, m)
										X = v[x]
										y = r[X]
										c = C(-59085)
										X = C(-59074)
										c = z[c]
										r = y
										y = k[X]
										m = { c(z) }
										c = { y(M(m)) }
										q = c[3]
										x = c[2]
										X = c[1]
										y = 9994545
									else
										p = C(-59081)
										v = k[p]
										p = C(-59097)
										Y = v[p]
										y = k[C(-59114)]
										p = G[s[1]]
										v = { Y(p) }
										g = { M(v) }
									end
								end
							end
						else
							if y < 15986299 then
								if y < 14635008 then
									if y < 14144813 then
										T = 938365
										u = 15909045308253
										m = C(-59116)
										c = u - T
										u = C(-59100)
										q = p(m, c)
										x = v[q]
										X = C(-59091)
										X = S[X]
										j = 424749
										Q = 27121398110265
										q = C(-59091)
										X = X(S, x)
										q = S[q]
										f = 8740525611769
										T = f - j
										c = p(u, T)
										j = 25602007324026
										a = -296873
										m = v[c]
										T = C(-59088)
										q = q(S, m)
										f = j - a
										u = p(T, f)
										m = C(-59091)
										j = 11816890085426
										c = v[u]
										u = C(-59099)
										m = q[m]
										m = m(q, c)
										f = -689461
										T = f + j
										c = p(u, T)
										j = 32506377287747
										f = 966142
										q = v[c]
										x = m[q]
										V = 559729
										u = C(-59092)
										T = f + j
										c = p(u, T)
										m = v[c]
										q = C(-59091)
										u = C(-59091)
										q = S[q]
										q = q(S, m)
										a = V + Q
										m = nil
										y = 16417782
										j = C(-59096)
										u = S[u]
										V = 12299013528266
										Q = -659956
										f = p(j, a)
										T = v[f]
										j = C(-59093)
										u = u(S, T)
										a = V - Q
										f = p(j, a)
										T = v[f]
										j = C(-59107)
										c = u[T]
										Q = 31550718330070
										V = 536049
										T = C(-59083)
										u = k[T]
										a = V + Q
										f = p(j, a)
										T = v[f]
										f = true
										u[T] = f
									else
										j = 443374
										g = #T
										a = 443374
										f = j - a
										y = g == f
										y = 8378112
									end
								else
									if y < 15427347 then
										y = nil
										v = y
										z = C(-59122)
										Y = N
										q = -760822
										u = -537521
										x = 31137
										r = C(-59081)
										g = nil
										p = g
										X = C(-59108)
										g = k[z]
										z = C(-59101)
										y = g[z]
										c = i()
										S = C(-59122)
										z = i()
										G[z] = y
										g = k[S]
										S = C(-59068)
										y = g[S]
										S = y
										g = k[r]
										r = C(-59097)
										y = g[r]
										g = k[X]
										X = C(-59090)
										r = y
										y = g[X]
										g = -31137
										X = y
										y = g + x
										f = 763449
										x = i()
										G[x] = y
										V = 436911
										g = 760824
										y = g + q
										j = 763448
										q = i()
										G[q] = y
										y = {}
										g = -537521
										m = y
										y = {}
										a = 551793
										G[c] = y
										y = g - u
										u = y
										y = {}
										g = f - j
										j = 552049
										f = j - a
										j = f
										a = -436910
										T = y
										f = a + V
										a = f
										y = 8609031
										f = 0
										V = a < f
										f = g - a
									else
										y = 9994545
										m = nil
										u = nil
										c = nil
									end
								end
							else
								if y < 16381602 then
									if y < 16089034 then
										j = 443374
										a = 443374
										g = #T
										f = j - a
										y = g == f
										y = y and 13017486 or 8378112
									else
										y = 11503455
									end
								else
									f = C(-59083)
									a = C(-59113)
									F = 34448835692761
									T = k[f]
									Q = 799865
									V = Q + F
									j = p(a, V)
									f = v[j]
									u = T[f]
									y = u and 9058580 or 8014581
								end
							end
						end
					end
				end
				y = #e
				return M(g)
			end, function(k, C)
				local M = p(C)
				local N = function(N, s, e, w)
					return y(k, { N, s, e, w }, C, M)
				end
				return N
			end, {}, function()
				v = 1 + v
				Y[v] = 1
				return v
			end
		return (S(15138254, {}))(M(g))
	end)(
		getfenv and getfenv() or _ENV,
		unpack or table[C(-59105)],
		newproxy,
		setmetatable,
		getmetatable,
		select,
		{ ... }
	)
end)(...)
